service/frontend/workflow_handler.go

```diff
@@ -3070,7 +3071,12 @@ func (wh *WorkflowHandler) validateStartWorkflowArgsForSchedule(
 		return errIDReusePolicyNotAllowed
 	}
 
-	if err := wh.validateSearchAttributes(startWorkflow.GetSearchAttributes(), namespaceName); err != nil {
+	// Unalias startWorkflow search attributes only for validation. Keep aliases in request.
+	unaliasedStartWorkflowSas, err := searchattribute.UnaliasFields(wh.saMapperProvider, startWorkflow.GetSearchAttributes(), namespaceName.String())
+	if err != nil {
+		return err
+	}
+	if err := wh.validateSearchAttributes(unaliasedStartWorkflowSas, namespaceName); err != nil {
```

Comment:
I actually don't like bellow lint message. I always do explicit error return. Because what it suggests looks like the function returns results of another function.

Reply:
Yeah, this contradicts the standard Go style of having the error path exit early and the happy path flow to the end of the function. Sometimes for short functions I might return a call directly but if there are other error returns I agree the last one should also be an error return for consistency